src/platform/assets/components/AssetBrowserModal.vue

```diff
@@ -21,13 +21,27 @@
     </template>
 
     <template #header>
-      <SearchBox
-        v-model="searchQuery"
-        :autofocus="true"
-        size="lg"
-        :placeholder="$t('assetBrowser.searchAssetsPlaceholder')"
-        class="max-w-96"
-      />
+      <div class="flex w-full items-center justify-between gap-2">
+        <SearchBox
+          v-model="searchQuery"
+          :autofocus="true"
+          size="lg"
+          :placeholder="$t('assetBrowser.searchAssetsPlaceholder')"
+          class="max-w-96"
+        />
+        <IconTextButton
+          v-if="isUploadButtonEnabled"
+          type="primary"
+          size="sm"
+          class="!bg-[var(--p-primary-color)] !text-[var(--p-primary-contrast-color)] !h-10"
```

Comment:
Should we build out the IconTextButton component to handle styling of colors?

Reply:
We explicitly set unstyled prop in the component so the styles are ignored by Primevue